src/base/bittorrent/torrentimpl.cpp

```diff
@@ -103,7 +104,14 @@ namespace
 
     QString toString(const lt::tcp::endpoint &ltTCPEndpoint)
     {
-        return QString::fromStdString((std::stringstream() << ltTCPEndpoint).str());
+        static QCache<lt::tcp::endpoint, QString> cache;
+
+        if (const QString *endpointName = cache.object(ltTCPEndpoint))
+            return *endpointName;
+
+        const QString endpointName = QString::fromStdString((std::stringstream() << ltTCPEndpoint).str());
```

Comment:
The object size of `std::ostringstream` is smaller: 376 vs 392 (`std::stringstream`). This was on linux.
```suggestion
        const QString endpointName = QString::fromStdString((std::ostringstream() << ltTCPEndpoint).str());
```

Reply:
Ok. Although it gives a very slight performance boost, especially since it has been used only a few times to fill the cache, I still changed it.